kanin/src/app/task.rs

```diff
@@ -79,7 +80,11 @@ where
             // Requests are handled and replied to concurrently.
             // This allows each handler task to process multiple requests at once.
             tasks.push(tokio::spawn(async move {
-                handle_request(req, handler, channel, should_reply).await;
+                let req_id = ReqId::extract(&mut req).await.expect("infallible");
```

Comment:
Actually I think if we're extracting a `req_id` for every request anyway, shouldn't we just extract it as part of `Request::new` and store it in the request? Then the `Extract` impl for `ReqId` can also be simplified to just take the stored `ReqId` from the request directly.

Or instead of storing it, at least just have a method on `Request` that gets the `ReqId` from the delivery inside, then the `Extract` impl could use that. But if we extract it every time anyway, maybe storing it isn't so bad? Any handler with an RPC client would be able to avoid extracting it twice then.

Reply:
Yeah, I def agree with that. Seems wild to extract it every time. Also a lot easier on the eyes to just write `req.req_id()`.